client/db.go

```diff
@@ -0,0 +1,529 @@
+// TODO(pmattis): ConditionalPut, DeleteRange
+
+package client
+
+import (
+	"encoding"
+	"fmt"
+	"log"
+	"net/url"
+	"strconv"
+	"time"
+
+	"github.com/cockroachdb/cockroach/proto"
+	"github.com/cockroachdb/cockroach/testutils"
+	gogoproto "github.com/gogo/protobuf/proto"
+)
+
+type byteSliceStringer []byte
+
+func (v byteSliceStringer) String() string {
+	if v == nil {
+		return "nil"
+	}
+	return string(v)
+}
+
+type int64Stringer int64
+
+func (v int64Stringer) String() string {
+	return strconv.FormatInt(int64(v), 10)
+}
+
+// KeyValue ...
+type KeyValue struct {
+	key       []byte
+	value     interface{}
+	timestamp int64
+}
+
+func (kv *KeyValue) String() string {
+	return string(kv.key) + ":" + kv.ValueStringer().String()
+}
+
+// Key ...
+func (kv *KeyValue) Key(k interface{}) error {
+	switch t := k.(type) {
+	case encoding.BinaryUnmarshaler:
+		return t.UnmarshalBinary(kv.key)
+	case *[]byte:
+		*t = kv.key
+	case *string:
+		*t = string(kv.key)
+	}
+	return fmt.Errorf("unable to unmarshal value: %T", k)
+}
+
+// KeyBytes ...
+func (kv *KeyValue) KeyBytes() []byte {
+	return kv.key
+}
+
+// Exists ...
+func (kv *KeyValue) Exists() bool {
+	return kv.value != nil
+}
+
+// Value ...
+func (kv *KeyValue) Value(v interface{}) error {
+	switch val := kv.value.(type) {
+	case []byte:
+		switch t := v.(type) {
+		case encoding.BinaryUnmarshaler:
+			return t.UnmarshalBinary(val)
+		case gogoproto.Message:
```

Comment:
What are we going to do about other protobuf implementations which people are more likely to use than gogoproto? It's a shame that protobuf doesn't support UnmarshalBinary. Am I missing why they couldn't / don't?